### PR TITLE
Add Email-Domain Namespace to Toggles

### PR DESCRIPTION
**Pull request by biyeun**
a little bit of a branch off of work done for
https://dimagi-dev.atlassian.net/browse/SAAS-11201

##### SUMMARY
This adds a new namespace for "email-domains" to our toggle setup. Basically, this checks if @<item> is present in the string passed to it in `enabled`, or if `request.user.username` or `request.user.email` has an email domain that matches `<item>`. I originally developed this as a solution to the hubspot data issue, but after speaking with @orangejenny I decided to go with an account-level flag that can only be adjusted in the accounting UI but this `email-domain` namespace still seemed like a better tool for implementing for certain features (like restricting what user accounts can create new domains) or using it for email communications.

##### FEATURE FLAG
new namespace for feature flags.

##### RISK ASSESSMENT / QA PLAN
Low risk and new tests.

### CODE REVIEW

**Review by millerdev — COMMENTED** [review]
Mostly looks good. The first comment is the only one that may be blocking, depending on the answer.

File: corehq/apps/toggle_ui/views.py [hunk 1]
```diff
@@ -238,7 +239,7 @@ def _call_save_fn_and_clear_cache(static_toggle, previously_enabled, currently_e
             domain = entry
             if static_toggle.save_fn is not None:
                 static_toggle.save_fn(domain, enabled)
-        else:
+        elif namespace != NAMESPACE_EMAIL_DOMAIN:
             # these are sent down with no namespace
             assert ':' not in entry, entry
             username = entry
```
[the remark this answers is not in the document]
Comment (biyeun) [reply corehq/apps/toggle_ui/views.py:245]: @orangejenny I see your name on the git blame for this line. do you know what the intention for username was? @millerdev I added the `elif` to escape the `assert`, which I think is the only functional part here.

**Comment by biyeun** [comment]
re-pinging this @dannyroberts @millerdev as i would like to get this out ASAP to be part of the solution to stop these "test emails" from project spaces causing our bounce rates to skyrocket

File: corehq/apps/toggle_ui/views.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/toggle_ui/views.py:251]: Going back to [this version](https://github.com/dimagi/commcare-hq/blame/88bf6335de08f9c30fdbe079f81e1bb7df2be416/corehq/apps/toggle_ui/views.py#L160), `username` used to be necessary to clear the cache. It looks like somewhere along the line, the cache-related behavior was removed from this function and the `username` reference should also have been removed.

**Review by millerdev — APPROVED** [review]

**Pull request merged**